File: EyeGuideAndroid/app/src/main/java/com/ZacharyZhang/eyeguide/data/model/AIResponse.kt
package com.ZacharyZhang.eyeguide.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SceneRequest(
    val image: String,
    @SerialName("detail_level") val detailLevel: String = "standard",
    val language: String = "zh-CN",
)

@Serializable
data class ReadTextRequest(
    val image: String,
    val language: String = "zh-CN",
)

@Serializable
data class ConversationEntry(
    val role: String,
    val text: String,
)

@Serializable
data class ConversationRequest(
    val message: String,
    val image: String? = null,
    @SerialName("conversation_history") val conversationHistory: List<ConversationEntry> = emptyList(),
    val language: String = "zh-CN",
)

@Serializable
data class FindObjectRequest(
    val image: String,
    @SerialName("target_object") val targetObject: String,
    val language: String = "zh-CN",
)

@Serializable
data class SocialRequest(
    val image: String,
    val language: String = "zh-CN",
)

@Serializable
data class GeminiPart(
    val text: String? = null,
)

@Serializable
data class GeminiContent(
    val parts: List<GeminiPart> = emptyList(),
)

@Serializable
data class GeminiCandidate(
    val content: GeminiContent? = null,
    val finishReason: String? = null,
)

@Serializable
data class GeminiResult(
    val candidates: List<GeminiCandidate> = emptyList(),
)

@Serializable
data class AIResultResponse(
    val result: GeminiResult? = null,
    val error: String? = null,
)

fun AIResultResponse.extractText(): String {
    if (error != null) return error
    val text = result?.candidates?.firstOrNull()?.content?.parts?.firstOrNull()?.text
    return text ?: "No response received"
}

File: EyeGuideiOS/EyeGuide/Models/AIResponse.swift
import Foundation

struct GeminiResponse: Codable, Sendable {
    let result: GeminiResult
}

struct GeminiResult: Codable, Sendable {
    let candidates: [GeminiCandidate]?
}

struct GeminiCandidate: Codable, Sendable {
    let content: GeminiContent?
    let finishReason: String?
}

struct GeminiContent: Codable, Sendable {
    let parts: [GeminiPart]?
    let role: String?
}

struct GeminiPart: Codable, Sendable {
    let text: String?
}

struct ConversationEntry: Codable, Sendable {
    let role: String
    let text: String
}

extension GeminiResponse {
    var textContent: String {
        guard let candidates = result.candidates,
              let firstCandidate = candidates.first,
              let content = firstCandidate.content,
              let parts = content.parts,
              let firstPart = parts.first,
              let text = firstPart.text else {
            return ""
        }
        return text
    }
}
